File: Flight Software/Check Fraud/check_fraud.java
import java.util.Scanner;
import java.util.Date; 


/* Check Fraud Launchpad Software
 *
 * https://github.com/besser435?tab=repositories
 */

public class check_fraud {
   public static boolean debugMode = false;
   public static int launchCountdown = 10;

   public static void main(String[] args) {
      mainMenu();
   }

   
   static void relay(int state) {
      if (debugMode == false){   //this is so im not deafened while working on the thing
         if (state == 0) {
            //relay off;
         }
         if (state == 1) {
            //relay on;
         }
      }
   }


   static void beeper(int state) {
      if (debugMode == false){   //this is so im not deafened while working on the thing
         if (state == 0) {
            //beeper off;
         }
         if (state == 1) {
            //beeper on;
         }
      }
   }
   

   static void ignition() {
      System.out.println("Relay On");
      //Thread.sleep(1000);

   }


   static void delayCountdown() {
      Scanner scnr = new Scanner(System.in);
      System.out.println("Are you sure you want to start the countdown? y/n ");

      
      //String whichOption = scnr.nextLine();
      String whichOption = scnr.next();
      //scnr.close();
      if (whichOption.equals("y")) {
         System.out.println("Starting Countdown. Press CTRL + C to cancel");
         //Thread.sleep(1000);
         for (int i = launchCountdown; i > 0; i--) {
            System.out.println(i);
         }
         



      } else {
         System.out.println("Countdown Cancelled");
         mainMenu();
      }
      scnr.close(); 
   }


   static void mainMenu() {
      Scanner scnr = new Scanner(System.in);
      System.out.println("Check Fraud Launchpad Software");
      System.out.println("1. Begin Ignition Countdown");
      System.out.println("2. Ignite Now");
      System.out.print("Enter an option: ");

      int whichOption = scnr.nextInt(); 
      //int whichOption = scnr.next();
      //scnr.close();
      switch (whichOption) {
         case 1:
            delayCountdown();
            break;
         case 2:
            //ignite();
            break;
         default:
            System.out.println("Invalid option");
            mainMenu();
            break;
      }
      scnr.close();
   }
}

File: Flight Software/Check Fraud/check_fraud.py

"""
Check Fraud
Launchpad software
This might be written in Java later, idk


https://github.com/besser435?tab=repositories
"""

version = "v0.2"
date = "November 2022" 

debug = 1

import time, sys, os, board
if debug == False: 
    import neopixel, board
    from gpiozero import DigitalOutputDevice
    from gpiozero import InputDevice

from colorama import init   # pip install colorama
init()
from colorama import Fore, Back, Style
init(autoreset=True)


# Setup
if debug == False: 
    # Neopixel
    led_neo = neopixel.NeoPixel(board.D12, 8)     # NeoPixel on D12
    # Beeper
    beeper = DigitalOutputDevice(21)
    # Ignition relay
    ign_relay = DigitalOutputDevice(21)


# options
#debug is above the imports
launch_countdown = 2
ignition_time = 1

# storage


def cc():   # shortens this long command to just cc()
    os.system("cls" if os.name == "nt" else "clear")    # clears terminal


def beeper_f(state):
    #global beeper
    if debug == False:
        if state == 1:
            beeper.on()
        elif state == 0:
            beeper.off()


def relay(state):
    #global ign_relay
    if debug == False:
        if state == 1:
            ign_relay.on()
        elif state == 0:
            ign_relay.off()
    else:
        print(Fore.LIGHTRED_EX + "(DEBUG) Ignition relay will be " + str(state))


def ignition():
    print(Fore.LIGHTGREEN_EX + "Relay on")
    #led_neo.fill((255, 0, 0))
    relay(1)
    time.sleep(ignition_time)

    relay(0)
    print(Fore.LIGHTGREEN_EX + "Relay off")
    #led_neo.fill((0, 0, 0))
    print()
    main_menu()


def countdown_led():
    for i in range:
        led_neo[i](255, 0, 0)
    

def delay_countdown():  # The type param is so other code can also use the countdown
    global launch_countdown
    #global LAUNCH_COUNTDOWN_INIT
    cc()

    confirm_countdown = input("Are you sure you want to start the countdown? y/n ")
    if confirm_countdown == "y":
        cc()




        countdown_time = input("How long should the countdown be? ")
        try:
            countdown_time = int(countdown_time)
            cc()
        except:
            print(Fore.RED + "Not a number! ")
            main_menu()
        launch_countdown = countdown_time
        #LAUNCH_COUNTDOWN_INIT = launch_countdown # delete this
        




        print(Fore.CYAN + "Starting Countdown. Press CTRL + C to cancel")   
        print()

        for i in range(launch_countdown):
            launch_countdown -= 1
            print(Fore.LIGHTYELLOW_EX + str(launch_countdown))
            #led_neo.fill((255, 0, 0))
            beeper_f(1)
            time.sleep(0.5)

            #led_neo.fill((0, 0, 0))
            beeper_f(0)
            time.sleep(0.5)

            #check if the connection between my pc and the Pi is active.
            #that way if the connection stops it automatically stops the countdown
            #this is incase we need to cancel the countdown
        #launch_countdown = LAUNCH_COUNTDOWN_INIT # reset the countdown
        ignition()
    else:
        cc()
        print(Fore.YELLOW + "Countdown aborted")
        main_menu()


def ignite_now():
    # once delay_countdown() is done, copy and paste that code here
    pass


def main_menu():    

    # print(f"Check Fraud Launchpad Software {version} {date}")
    print("Check Fraud Launchpad Software")
    print(version)
    print(date)
    if debug == True: print(Fore.LIGHTRED_EX + "DEBUG MODE")
    print(Fore.CYAN + "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~")
    print(Fore.LIGHTGREEN_EX + "Options:")
    print("1: Begin Ignition Countdown")
    print("2: Ignite Now")
    #print("3: ")
    print("Q: Quit")
  
    which_option = input("What would you like to do? ")

    if "1" is which_option:
        delay_countdown()
        cc()
        
    elif "2" is which_option:
        cc()

    elif "3" is which_option:
        cc()

    elif "q" is which_option:
        sys.exit()
    
    elif "a" is which_option:  # shortcut to current work
        pass 

    else:
        cc()
        print(Fore.RED + "Invalid input")
        main_menu()
main_menu()




